src/Middleware/HeaderPropagation/src/HeaderPropagationOptions.cs

```diff
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.HeaderPropagation
+{
+    public class HeaderPropagationOptions
+    {
+        public IList<HeaderPropagationEntry> Headers { get; set; } = new List<HeaderPropagationEntry>();
+    }
+}
```

Comment:
The drawback of using a list is that you can express duplicates (same outbound name). I'm not sure that it's really a problem that needs to be solved.

Reply:
I had decided to change it to a dictionary, as overriding arrays in config is always a bit messy. Also the ordering doesn't add much value imo. However I used the input name as key so it doesn't address the potential issue you raised.
The reason why I used the input name is that the input name is the one relevant in the middleware. The output name is only used by the Handler, is optional and I believe it should be moved to be moved to the handler configuration as part as handling the ability to specify which handler to include per client (in a following PR).